### ISSUE TITLE
TEST_DEPENDS use scenario is missing

### ISSUE THREAD

**Issue by jvegh** [cmake-basis/BASIS#604]
From the documentation it is not quite clear, what is TEST_DEPENDS for, and how it can be used. Probably it is for testing, and follows the same logic as DEPENDS, but I cannot conclude from the docs how to use it.

**Comment by schuhschuh** [comment]
Yes, these should be documented in the manual. For now, please refer to the API documentation. In this case the `basis_project` function of which `TEST_DEPENDS` is an option of.

You find the reference here:
http://cmake-basis.github.io/apidoc/latest/group__CMakeAPI.html#gad82d479d14499d09c5aeda3af646b9f6

EDIT: This page is embedded in the online documentation at:
http://cmake-basis.github.io/apidoc/modules.html
following the link to "Public CMake Interface". Unfortunately, the use of `iframe` hides the actual link above...

**Comment by schuhschuh** [comment]
In case the API documentation is not clear enough, it only says:

> List of dependencies, i.e., either names of other BASIS (sub)projects or names of external packages which are only required by the tests.

`TEST_DEPENDS` and `OPTIONAL_TEST_DEPENDS` are only looked for when `BUILD_TESTING` flag is set to `ON`, i.e., when the source files in the `TESTING_DIR` (default: `test/`) are included in the build.

`TOOLS_DEPENDS` and `OPTIONAL_TOOLS_DEPENDS` are only looked for when `BUILD_APPLICATIONS` is set to `ON`, i.e., when the source files in the `TOOLS_DIR` (default: `tools/`) are included in the build.
